connection.php

```diff
@@ -0,0 +1,1028 @@
+    {
+        foreach ($data as $key => $value) {
+            if (property_exists($object, $key)) {
+                $object->$key = $value;
+            }
+        }
+        return $object;
+    }
+
+
+    public function find($value = null, $key = null)
+    {
+        $this->checkClass();
+        $properties = $this->getProperties();
+        $this->from($properties->table, false);
+        if ($value !== null) {
+            if (is_int($value) && property_exists($properties, 'id_field')) {
+                $this->where($properties->id_field, $value);
+            } else if (is_string($value) && property_exists($properties, 'name_field')) {
+                $this->where($properties->name_field, $value);
+            } else if (is_array($value)) {
+                $this->where($value);
+            }
+        }
+        if (empty($this->sql)) {
+            $this->select();
+        }
+        $data = $this->many($key);
+        $objects = array();
+        foreach ($data as $row) {
+            $objects[] = $this->load(new $this->class, $row);
+        }
+        return (sizeof($objects) == 1) ? $objects[0] : $objects;
+    }
+
+    public function save($object, array $fields = null)
+    {
+        $this->using($object);
+        $properties = $this->getProperties();
+        $this->from($properties->table);
+        $data = get_object_vars($object);
+        $id = $object->{$properties->id_field};
+        unset($data[$properties->id_field]);
+        if ($id === null) {
+            $this->insert($data)
+                ->execute();
+            $object->{$properties->id_field} = $this->insert_id;
+        } else {
+            if ($fields !== null) {
+                $keys = array_flip($fields);
+                $data = array_intersect_key($data, $keys);
+            }
+            $this->where($properties->id_field, $id)
+                ->update($data)
+                ->execute();
+        }
+        return $this->class;
+    }
+
+    public function remove($object)
+    {
+        $this->using($object);
+        $properties = $this->getProperties();
+        $this->from($properties->table);
+        $id = $object->{$properties->id_field};
+        if ($id !== null) {
+            $this->where($properties->id_field, $id)
+                ->delete()
+                ->execute();
+        }
+    }
+
+
+    public function getProperties()
+    {
+        static $properties = array();
+        if (!$this->class) return array();
+        if (!isset($properties[$this->class])) {
+            static $defaults = array(
+                'table' => null,
+                'id_field' => null,
+                'name_field' => null
+            );
+            $reflection = new ReflectionClass($this->class);
+            $config = $reflection->getStaticProperties();
+            $properties[$this->class] = (object)array_merge($defaults, $config);
+        }
+        return $properties[$this->class];
+    }
+}
```

Comment:
@araksia This looooong class isn't a db connection manager rather than a sql/command/query builder. It isn't quite what the test was asking,